src/query/point/point_triangle.rs

```diff
@@ -216,23 +215,73 @@ impl PointQueryWithLocation for Triangle {
                 );
             }
             ProjectionInfo::OnFace(face_side, va, vb, vc) => {
-                // Voronoï region of the face.
-                if DIM != 2 {
-                    // NOTE: in some cases, numerical instability
-                    // may result in the denominator being zero
-                    // when the triangle is nearly degenerate.
-                    if va + vb + vc != 0.0 {
-                        let denom = 1.0 / (va + vb + vc);
-                        let v = vb * denom;
-                        let w = vc * denom;
-                        let bcoords = [1.0 - v - w, v, w];
-                        let res = a + ab * v + ac * w;
-
+                let inv_denom = va + vb + vc;
+                if inv_denom == 0.0 {
+                    // The triangle is degenerate (the three points are collinear).
+                    // So we find the longest segment and project point on it.
+
+                    // TODO: this can probably be optimized using data we already have ?
```

Comment:
I'm out of my depth to find the correct data to operate on, but as it's quite the edge case I'd advocate to merge as-is and consider improving it in another PR.